core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -323,7 +323,12 @@ struct math_function_name;
 // https://www.gnu.org/software/libc/manual/html_node/Errors-in-Math-Functions.html
 // For now 1s largely seem to work ...
 DEFINE_UNARY_FUNCTION_EVAL(exp, 2);
+#ifdef KOKKOS_COMPILER_NVHPC  // FIXME WORKAROUND exp2 not device callable,
+                              // workaround computes it via exp
+DEFINE_UNARY_FUNCTION_EVAL(exp2, 30);
```

Comment:
Would you need less ULP when you use `double` also for the `float` overload? Or do you also have problems with the non-float overloads?

Reply:
its less of a problem with double, but you need more than the 2, because there is more rounding going on in the workaround version.